bindings/python/util/demangle.hpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * ghex-org
+ *
+ * Copyright (c) 2014-2023, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <cxxabi.h>
+#include <string>
+#include <memory>
+#include <typeinfo>
+#include <stdexcept>
+
+namespace pyghex
+{
+namespace util
+{
+
+template<typename T>
+std::string
+demangle()
+{
+    int                                    status = 0;
+    std::unique_ptr<char, void (*)(void*)> res{
+        abi::__cxa_demangle(typeid(T).name(), nullptr, nullptr, &status), std::free};
```

Comment:
What compilers support this? I think clang does, right? But I'm not sure about intel or Cray. This may limit the support of GHEX for some compilers. Fine for now, but we may document this in the readme?